src/visualization/CMakeLists.txt

```diff
@@ -12,6 +17,154 @@ if(APPLE)
       COMMAND xcodebuild -project "${CMAKE_CURRENT_SOURCE_DIR}/Turi Create Visualization.xcodeproj/" -configuration ${CMAKE_BUILD_TYPE} SYMROOT=${CMAKE_CURRENT_BINARY_DIR}
       COMMENT "Building visualization client via xcodebuild"
   )
+  else()
+    make_empty_library(visualization_client)
+  endif()
+elseif(LINUX)
+  if(${TC_BUILD_VISUALIZATION_CLIENT})
+
+    find_program(NPM npm)
+    if(NOT NPM)
+      message(FATAL_ERROR "npm not found. Node.js 6.x or later with npm is required to build Turi Create.")
+    endif()
+
+
+
+    include(ExternalProject)
+
+    ExternalProject_Add(
+        cef
+        PREFIX "cef"
+        URL http://opensource.spotify.com/cefbuilds/cef_binary_3.3282.1731.gfc9a4fa_linux64_minimal.tar.bz2
+        URL_HASH SHA1=4eeeabee642bfe68f561566e0f249f86afaf6634
+        DOWNLOAD_DIR ${CMAKE_CURRENT_BINARY_DIR}/deps
+        SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/deps/cef/src/cef
+        BUILD_IN_SOURCE 1
+        CONFIGURE_COMMAND ${CMAKE_COMMAND} .
+        BUILD_COMMAND "make"
+        INSTALL_COMMAND ""
+    )
+
+
+    set(CEF_LIB_SO ${CMAKE_CURRENT_BINARY_DIR}/deps/cef/src/cef/Release/libcef.so)
+    set(CEF_RESOURCES ${CMAKE_CURRENT_BINARY_DIR}/deps/cef/src/cef/Resources/*)
+
+    set(TC_VIZ_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/TcViz)
+    set(TC_VIZ_HTML_DIRECTORY ${TC_VIZ_DIRECTORY}/html)
+
+    set(CEF_RESOURCES_LOCALE ${CMAKE_CURRENT_BINARY_DIR}/deps/cef/src/cef/Resources/locales/*)
+    set(TC_VIZ_LOCALES_DIRECTORY ${TC_VIZ_DIRECTORY}/locales)
+
+    set(TCVIZ_SRCS
+      src/layer.cpp
+      src/layer.h
```

Comment:
You shouldn't need to include .h files in the list of sources. Only .cpp files should be listed as part of the build (CMake is smart enough to gather the include dependencies and pick up changes to headers on incremental builds).

Reply:
noted, changed